### PR TITLE
✨ Disallow post swap entry point and adapter contract calls

### LINKED ISSUE #7: Oak Security Audit - Finding 5

### PR DESCRIPTION
**Pull request by NotJeremyLiu**
Closes #7

### CODE REVIEW

**Review by jaybxyz — APPROVED** [review]
LGTM. Let's wait others to verify the revision and approve.

**Comment by NotJeremyLiu** [comment]
@sushiwushi Could you review this PR / approve when you get a chance. I think once this is approved I can begin merging all the PRs in for the audit findings. Please inform me if I'm mistaken, thanks!

**Comment by sushiwushi** [comment]
Hi @NotJeremyLiu, the fix LGTM. Sorry for the delay in checking the fixes!

**Pull request merged**
